### ISSUE TITLE
PartialSearchHelperUsage doesn't detect PartialSearch class usage

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#365]
I have no idea why it was ever done this way, but I ran across a few uses of Partial Search like this:

```ruby
        ::Chef::PartialSearch.new.search(search_key, query, :keys => partial_search_keys, :sort => sort_key) do |config|
          RealityForge::AttributeTools.deep_merge(mash, input_path, config['output'])
        end
```

**Comment by tas50** [comment]
Fixed with its own cop in #367
